lib/zlib.js

```diff
@@ -121,8 +127,8 @@ function zlibBuffer(engine, buffer, callback) {
 function zlibBufferSync(engine, buffer) {
   if (typeof buffer === 'string')
     buffer = Buffer.from(buffer);
-  if (!(buffer instanceof Buffer))
-    throw new TypeError('Not a string or buffer');
+  else if (!isUint8Array(buffer))
+    throw new TypeError('Not a string, a Buffer, or an Uint8Array');
```

Comment:
As long as the error message is being changed, it would be better to make it more consistent ...

e.g.
```
"buffer" must be a string, Buffer, or Uint8Array
```